pilot/pkg/features/pilot.go

```diff
@@ -406,6 +406,10 @@ var (
 		"If this is set to true, support for Kubernetes gateway-api (github.com/kubernetes-sigs/gateway-api) will "+
 			" be enabled. In addition to this being enabled, the gateway-api CRDs need to be installed.").Get()
 
+	EnableAlphaGatewayAPI = env.Register("PILOT_ENABLE_ALPHA_GATEWAY_API", false,
```

Comment:
Yes, it's either that or they have an outage when someone decides to pull the rug on the alpha version. I am certain that will happen at least on GKE when they move to beta

Reply:
> An alternative could be to read the actual CRD to check it has the version we want. However, this is not safe -- a user may silently stop reading critical configurations.

Could you expand on who the "user" is in this scenario and what configurations they're going to cease reading. FWIW, this was what I envisioned the implementation looking like

Reply:
Scenario I am concerned about:
* Istio user creates an Route alpha. They rely on this for prod.
* Infra admin installs CRD with only Route beta.
* Istio sees there is no Alpha CRD and ignores it
* User now silently has all their config dropped

Reply:
Maybe I'm missing something, but how would the alpha Route CR exist if the admin installs/upgrades to the beta versions? etcd can only have one version

Reply:
Hmm.. I am actually not sure what happens if you go from having only alpha version to only beta version. Does k8s even allow it?

It can still break if you go from alpha -> {alpha,beta} -> beta for sure though. The storage version is beta in that case.

Reply:
Ah because the alpha version would be deleted; that's the user config. Got it.